crates/js-component-bindgen/src/ts_bindgen.rs

```diff
@@ -217,36 +222,72 @@ pub fn ts_bindgen(
+                    /**
+                     * Instantiates this component with the provided imports and
+                     * returns a map of all the exports of the component.
+                     *
+                     * This function is intended to be similar to the
+                     * `WebAssembly.instantiate` function. The second `imports`
+                     * argument is the \"import object\" for wasm, except here it
+                     * uses component-model-layer types instead of core wasm
+                     * integers/numbers/etc.
+                     *
+                     * The first argument to this function, `getCoreModule`, is
+                     * used to compile core wasm modules within the component.
+                     * Components are composed of core wasm modules and this callback
+                     * will be invoked per core wasm module. The caller of this
+                     * function is responsible for reading the core wasm module
+                     * identified by `path` and returning its compiled
+                     * `WebAssembly.Module` object. This would use `compileStreaming`
+                     * on the web, for example.
+                     */
+                    export function instantiate(
+                        getCoreModule: (path: string) => Promise<WebAssembly.Module>,
+                        imports: ImportObject,
+                        instantiateCore?: (module: WebAssembly.Module, imports: Record<string, any>) => Promise<WebAssembly.Instance>
+                    ): Promise<{camel}>;
+                ",
+            )
+        }
+
+        Some(InstantiationMode::Sync) => {
+            uwriteln!(
+                bindgen.src,
+                "
+                    /**
+                     * Instantiates this component with the provided imports and
+                     * returns a map of all the exports of the component.
+                     *
+                     * This function is intended to be similar to the
+                     * `WebAssembly.Instantiate` constructor. The second `imports`
+                     * argument is the \"import object\" for wasm, except here it
+                     * uses component-model-layer types instead of core wasm
+                     * integers/numbers/etc.
+                     *
+                     * The first argument to this function, `getCoreModule`, is
+                     * used to compile core wasm modules within the component.
+                     * Components are composed of core wasm modules and this callback
+                     * will be invoked per core wasm module. The caller of this
+                     * function is responsible for reading the core wasm module
+                     * identified by `path` and returning its compiled
+                     * `WebAssembly.Module` object. This would use the
+                     * `WebAssembly.Module` constructor on the web, for example.
```

Comment:
In the case of synchronous compilation, `getCoreModule` would have to be assumed to be cached. So perhaps we should update the comment to something like:

```suggestion
                     * The first argument to this function, `getCoreModule`, is
                     * used to get the compiled core wasm modules within the component.
                     * Components are composed of core wasm modules and this callback
                     * will be invoked per core wasm module. The caller of this
                     * function is responsible for reading the core wasm module
                     * identified by `path` and returning its compiled
                     * `WebAssembly.Module` object. This should ideally be an 
                     * already-cached cached compilation instead of created by the 
                     * `WebAssembly.Module` constructor on the web.
```

Reply:
I'll land it with this rewording soon, unless you have any further feedback.

Reply:
I'm fine with the rewording :-). Thanks!